### PR TITLE
cmd/snap: improve output when snaps were found in a section or the section is invalid

### PR DESCRIPTION
**Pull request by bboozzoo**
Be more verbose when searching a snap in a section did not return anything or
the section does not exist, thus also no results are returned for the query.
```
$ snap find --section=foobar foo
No matching snaps for "foo"
$ snap find --section=foobar
No matching snaps for ""
$ snap find --section=video foo
No matching snaps for "foo"
```
becomes:
```
$ snap find --section=foobar foo                                                                                                                       
error: No matching section "foobar", use --section to list existing sections
$ snap find --section=foobar                                                                                                                           
error: No matching section "foobar", use --section to list existing sections
$ snap find --section=video foo
No matching snaps for "foo" in section "video"
```

### CODE REVIEW

File: cmd/snap/cmd_find.go [hunk 1]
```diff
@@ -154,6 +155,18 @@ func (x *cmdFind) Execute(args []string) error {
 	}
 
 	cli := Client()
+
+	if x.Section != "" && x.Section != "featured" {
+		sections, err := cli.Sections()
```
Comment (mvo5) [inline cmd/snap/cmd_find.go]: I wonder if this is an extra roundtrip to the store? And if so, if we can get the store to provide error information like "no-such-group" for us?

Comment (bboozzoo) [reply cmd/snap/cmd_find.go]: Digging deeper, I think we should be able to save on the roundtrip and use the cached sections files that's already there for the completion support.

File: cmd/snap/cmd_find.go [hunk 2]
```diff
@@ -154,6 +176,18 @@ func (x *cmdFind) Execute(args []string) error {
 	}
 
 	cli := Client()
+
+	if x.Section != "" && x.Section != "featured" {
+		sections, err := getSections()
+		if err != nil {
+			return err
+		}
+		if !strutil.ListContains(sections, string(x.Section)) {
```
Comment (chipaca) [inline cmd/snap/cmd_find.go]: Given that the sections file is sorted, I saw this and thought “could use `strutil.SortedListContains`”, but then I realised that the sections are sorted in `refreshCatalogs`, not in `getSections`. I'll move the sort and switch this to the more efficient search in a followup.

**Comment by chipaca** [comment]
Was going to add, but forgot: nice work, thank you.

**Pull request merged**